### ISSUE TITLE
[bug | question] The "docker" server crashed

### ISSUE THREAD

**Issue by onichandame** [josa42/coc-docker#45]
# Observation

When editing a Dockerfile, the language server could not start. Maybe it is not a bug but me missing something here. Some guidance to debug would be appreciated.

# Diagnostics

`:message`:

```text
[coc.nvim] The "docker" server crashed 5 times in the last 3 minutes. The server will not be restarted.
```

`:CocInfo`:

```text
## versions    
                                                                                                                                                                                                           
vim version: NVIM v0.4.3    
node version: v14.5.0    
coc.nvim version: 0.0.79-365c63e12f    
coc.nvim directory: /root/.config/nvim/plugged/coc.nvim    
term: xterm-256color    
platform: linux    
                                                                                                           
## Log of coc.nvim    
                                                                                                           
2020-12-17T03:11:57.304 WARN (pid:78) [attach] - Plugin not ready when received "highlight" []    
2020-12-17T03:11:59.813 INFO (pid:78) [services] - registered service "docker"    
2020-12-17T03:11:59.816 INFO (pid:78) [services] - dockerfile-language-server-nodejs state change: stopped => starting    
2020-12-17T03:12:00.310 INFO (pid:78) [language-client-index] - docker started with 89    
2020-12-17T03:12:00.874 INFO (pid:78) [plugin] - coc.nvim 0.0.79-365c63e12f initialized with node: v14.5.0 after 3531ms    
2020-12-17T03:12:00.935 INFO (pid:78) [services] - dockerfile-language-server-nodejs state change: starting => stopped    
2020-12-17T03:12:00.935 INFO (pid:78) [services] - dockerfile-language-server-nodejs state change: stopped => starting    
2020-12-17T03:12:00.977 INFO (pid:78) [language-client-index] - docker started with 96    
2020-12-17T03:12:01.119 INFO (pid:78) [services] - dockerfile-language-server-nodejs state change: starting => stopped    
2020-12-17T03:12:01.119 INFO (pid:78) [services] - dockerfile-language-server-nodejs state change: stopped => starting    
2020-12-17T03:12:01.135 INFO (pid:78) [language-client-index] - docker started with 104    
2020-12-17T03:12:01.242 INFO (pid:78) [services] - dockerfile-language-server-nodejs state change: starting => stopped    
2020-12-17T03:12:01.242 INFO (pid:78) [services] - dockerfile-language-server-nodejs state change: stopped => starting    
2020-12-17T03:12:01.259 INFO (pid:78) [language-client-index] - docker started with 112    
2020-12-17T03:12:01.365 INFO (pid:78) [services] - dockerfile-language-server-nodejs state change: starting => stopped    
2020-12-17T03:12:01.365 INFO (pid:78) [services] - dockerfile-language-server-nodejs state change: stopped => starting    
2020-12-17T03:12:01.383 INFO (pid:78) [language-client-index] - docker started with 119    
2020-12-17T03:12:01.492 INFO (pid:78) [services] - dockerfile-language-server-nodejs state change: starting => stopped
```

# Environment

- node: shown above
- vim: shown above
- coc-docker: 0.4.0
- OS: docker container `docker:dind`

**Comment by 0x8f701** [comment]
coc doesn't support LSP 3.16 yet

**Comment by onichandame** [comment]
Found the problems. There are 2 as far as I have seen:

1. the resolved path to the language server presumes an unreliable directory structure. Fixed by #46 
2. as pointed out by @GopherJ , LSP 3.16 still lacks support. [This PR](https://github.com/neoclide/coc.nvim/pull/2756#issue-544103205) tracks the effort to make LSP 3.16 compatible.
